runtime/src/core/closures.js

```diff
@@ -1,17 +1,16 @@
-import { managedMemory, grainModule } from "../runtime";
 import { grainToJSVal, JSToGrainVal } from "../utils/utils";
 
 export class GrainClosure {
   constructor(loc, runtime) {
-    const view = managedMemory.view;
+    const view = runtime.managedMemory.view;
 
     this.loc = loc;
     this.runtime = runtime;
     this.arity = view[loc];
-    this.ptr = view[loc + 1];
-    this.closureSize = view[loc + 2];
-    this.closureElts = view.slice(loc + 3, loc + 3 + this.closureSize);
-    this.func = grainModule.instance.exports["GRAIN$LAM_" + this.ptr];
+    this.ptr = view[loc + 2];
+    this.closureSize = view[loc + 3];
+    this.closureElts = view.slice(loc + 4, loc + 4 + this.closureSize);
+    this.func = runtime.table.get(this.ptr);
```

Comment:
Just for my understanding, can you explain a little more about this change?

Reply:
The previous method here worked back before we had separate compilation, since there was only one Grain module being loaded and we exported all of the functions in the module. Now of course there's multiple modules and we only export the closures. This grabs the table index from the closure, and then just looks up the function in the global webassembly table. Pretty neat, and makes this more likely to work if we decide that we want to try to fix this.